include/drivers/qspi.h

```diff
@@ -0,0 +1,660 @@
+				u32_t start_address,
+				u32_t length);
+
+
+/**
+ * @typedef qspi_api_write_async
+ * @brief Callback API for asynchronous write operation
+ * See qspi_write_async() for argument descriptions
+ */
+typedef int (*qspi_api_write_async)(struct device *dev,
+				const struct qspi_buf *tx_buf,
+				u32_t address,
+				struct k_poll_signal *async);
+
+/**
+ * @typedef qspi_api_read_async
+ * @brief Callback API for asynchronous read operation
+ * See qspi_read_async() for argument descriptions
+ */
+typedef int (*qspi_api_read_async)(struct device *dev,
+				const struct qspi_buf *rx_buf,
+				u32_t address,
+				struct k_poll_signal *async);
+
+/**
+ * @typedef qspi_api_send_cmd_async
+ * @brief Callback API for asynchronous custom command send operation
+ * See qspi_send_cmd_async() for argument descriptions
+ */
+typedef int (*qspi_api_send_cmd_async)(struct device *dev,
+				const struct qspi_cmd *cmd,
+				struct k_poll_signal *async);
+
+/**
+ * @typedef qspi_api_erase_async
+ * @brief Callback API for asynchronous erase operation
+ * See qspi_erase_async() for argument descriptions
+ */
+typedef int (*qspi_api_erase_async)(struct device *dev,
+				u32_t start_address,
+				u32_t length,
+				struct k_poll_signal *async);
+
+/**
+ * @brief QSPI driver API
+ * This is the mandatory API any QSPI driver needs to expose.
+ */
+struct qspi_driver_api {
+	qspi_api_configure configure;
+	qspi_api_write write;
+	qspi_api_read read;
+	qspi_api_send_cmd send_cmd;
+	qspi_api_erase erase;
+#ifdef CONFIG_QSPI_ASYNC
+	qspi_api_write_async write_async;
+	qspi_api_read_async read_async;
+	qspi_api_send_cmd_async send_cmd_async;
+	qspi_api_erase_async erase_async;
+#endif /* CONFIG_QSPI_ASYNC */
+};
+/**
+ * @endcond
+ */
+
+
+/**
+ * @brief Configures QSPI driver to desired memory.
+ *
+ * Note: This function is synchronous.
+ * If user want to configure QSPI flash memory for usage with quad data lines,
+ * one must set Quad Enble (QE) bit in Status Register. To do so use
+ * qspi_send_cmd function and Write Status Register (WRSR) command.
+ *
+ * @param dev Pointer to the device structure for the driver instance
+ * @param config Pointer to the configuration structure.
+ * See qspi_config structure for detailed description
+ *
+ * @retval  0 in case of success
+ * @retval	-ENXIO No such device or address
+ * @retval	-EINVAL invalid input parameter
+ * @retval	-EBUSY device busy
+ */
+__syscall int qspi_configure(struct device *dev,
+				const struct qspi_config *config);
+
+static inline int z_impl_qspi_configure(struct device *dev,
+				const struct qspi_config *config)
+{
+	const struct qspi_driver_api *api =
+		(const struct qspi_driver_api *)dev->driver_api;
+
+	return api->configure(dev, config);
+}
+
+
+/**
+ * @brief Writes desired amount of data to the external flash memory.
+ *
+ * Note: This function is synchronous.
+ * Driver that realizes write functionality has to:
+ *  1. Check if memory is ready for operation - using RDSR
```

Comment:
What's the advantage of requiring the driver do this instead of the JEDEC driver above it?